src/Test/Html/Query.elm

```diff
@@ -314,6 +315,27 @@ has selectors (Internal.Single showTrace query) =
         |> failWithQuery showTrace ("Query.has " ++ Internal.joinAsList selectorToString selectors) query
 
 
+{-| Expect the element to **not** match all of the given selectors.
+
+    import Html exposing (div, ul, li)
+    import Test.Html.Query as Query
+    import Test exposing (test)
+    import Test.Html.Selector exposing (tag, class)
+
+
+    test "The div element has no progress-bar class" <|
+        \() ->
+            div  []
+                |> Query.fromHtml
+                |> Query.find [ tag "div" ]
+                |> Query.hasNot [ tag "div", class "progress-bar" ]
+-}
+hasNot : List Selector -> Single -> Expectation
+hasNot selectors (Internal.Single showTrace query) =
+    Internal.hasNot selectors query
+        |> failWithQuery showTrace ("Query.hasNot " ++ Internal.joinAsList selectorToString selectors) query
```

Comment:
This is pretty hard to read - worth sticking `"Query.hasNot " ++ Internal.joinAsList selectorToString selectors` in a let binding